tests/e2e_tests/integration/07-saml.spec.ts

```diff
@@ -0,0 +1,183 @@
+      await expect(find_results_acs).toHaveText(expected_acs_url);
+      console.log(`got span with acs url: ${await find_results_acs.count()}`)
+
+      const expected_sp_meta_url=`${baseUrl}api/management/v1/useradm/sso/sp/metadata/${id}`;
+      const find_results_sp_meta = await page.locator('span',{'hasText' : expected_sp_meta_url})
+      await expect(find_results_sp_meta).toHaveText(expected_sp_meta_url);
+      console.log(`got span with SP metadata url: ${await find_results_sp_meta.count()}`)
+
+      const sp_meta_response = await axios({
+        url: `${baseUrl}api/management/v1/useradm/sso/sp/metadata/${id}`,
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${jwt}`,
+          'Content-Type': 'application/json'
+        },
+        httpsAgent: new https.Agent({
+          rejectUnauthorized: false
+        })
+      });
+
+      if (sp_meta_response.status !== 200) {
+        throw 'expected 201';
+      }
+
+      const service_provider_metadata=sp_meta_response.data.replaceAll('Signed="true"','Signed="false"')
+      fs.writeFileSync('fixtures/service_provider_metadata.xml', service_provider_metadata);
+
+      await page.goto('https://samltest.id/upload.php');
+      await page.waitForSelector('input[id=uploader]');
+
+      // Click input[name="userfile"]
+      await page.locator('input[name="userfile"]').click();
+      // Upload m
+      await page.locator('input[name="userfile"]').setInputFiles('fixtures/service_provider_metadata.xml');
+      // Click input:has-text("Upload")
+      await page.locator('input:has-text("Upload")').click();
+      await expect(page).toHaveURL('https://samltest.id/upload.php');
+
+      console.log('uploaded file, making screen shot, after waiting 4s');
+      await page.waitForTimeout(timeout_4s);
+      await page.screenshot({path: 'saml-uploaded.png'});
+
+      // Go to https://staging.hosted.mender.io/ui/settings/user-management
+      await page.goto(`${baseUrl}ui/settings/user-management`);
+      // Click text=Create new user
+      await page.locator('text=Create new user').click();
+      // Click [placeholder="Email"]
+      await page.locator('[placeholder="Email"]').click();
+      // Fill [placeholder="Email"]
+      await page.locator('[placeholder="Email"]').fill(saml_settings['email']);
+      // Click text=Create user
+      await page.locator('text=Create user').click();
+      await page.waitForTimeout(timeout_1s);
+      await expect(page.locator(`text=${saml_settings['email']}`)).toHaveText(saml_settings['email']);
+
+      console.log(`logging in via ${baseUrl}api/management/v1/useradm/auth/sso/${id}/login`);
+      // enter valid username and password
+      await page.goto(`${baseUrl}api/management/v1/useradm/auth/sso/${id}/login`);
+      await page.waitForTimeout(timeout_4s);
+      await page.screenshot({path: 'saml-redirected.png'});
+
+      // Double click text=morty
+      await page.locator('text=morty').dblclick();
```

Comment:
what does the double click here do?

Reply:
all it does is: it does not hurt. :) removed.

Reply:
keep in mind: every interaction that is not required, is still executed and the completion awaited, so execution time...

Reply:
point taken. this was me being a bit fast on the trigger when I recorded the session, and the forgot to remove.